tests/shell/shell_recorder/README.md

```diff
@@ -1,28 +1,31 @@
 # Shell Recorder
 
+The Shell Recorder is a test tool that checks the output and return values of  shell commands such as `kdb`. It allows you to write unit and  regression tests. The Shell Recorder also records the result of tests in so-called protocol files. These protocol files can be used as basis for new Shell Recorder tests.
```

Comment:
Please use line breaks to make commenting easier. Mentioning the protocol files without saying what they are and how to use them for new tests is not ideal.

Maybe directly forward to tutorial describing the markdown shell recorder here.